src/app/dashboard/layout.tsx

```diff
@@ -26,6 +30,12 @@ export default function DashboardLayout({
     }
   });
 
+  React.useLayoutEffect(() => {
+    if (!isLoading && !isAuthenticated) {
```

Comment:
make sure we are not loading here or we will get a redirect to user/login even when we are auth in the first page load (because we still didn't retrieve the state)